local-cli/server/util/attachHMRServer.js

```diff
@@ -16,11 +16,254 @@ const url = require('url');
  * Hot Module Replacement updates to the simulator.
  */
 function attachHMRServer({httpServer, path, packagerServer}) {
-  let client = null;
 
-  function disconnect() {
-    client = null;
-    packagerServer.setHMRFileChangeListener(null);
+  let clients = [];
+  let clientsPerBundleEntry = {};
+
+  let platformCache = {
```

Comment:
> except rare case on which you have both the simulator and a device connected to the Packager

I was imagining my own use case, which is testing on different screen sizes at the same time, by connecting an iPhone and an iPad for example, or an android phone and a tablet, or all of the above! :smile: 
This would effectively cut the processing time by half in that specific case.

On the other hand, I definitely agree with your approach, this should be isolated, which would lead to better error handling, and more maintainability. 

I think the tradeoff in speed is more than acceptable, especially since my use case should be pretty rare, most people would only want to develop on android and iOS simultaneously, and this complexity wouldn't speed that up at all.